src/net/rpc.rs

```diff
@@ -3814,6 +3827,24 @@ mod test {
         mempool_tx.commit().unwrap();
         peer_2.mempool.replace(mempool);
 
+        let mut peer_1_sortdb = peer_1.sortdb.take().unwrap();
+        let mut peer_1_stacks_node = peer_1.stacks_node.take().unwrap();
+        let _ = peer_1
+            .network
+            .refresh_burnchain_view(&peer_1_sortdb, &peer_1_stacks_node.chainstate, false)
```

Comment:
why did you add calls to `refresh_burnchain_view` here?

Reply:
The absence of this call was causing the `.canonical_stacks_tip_height` values to not propagate.  Adding this in fixes it.